diff -ruN gcc-10.2.0/libcpp/expr.c gcc-10.2.0-patched/libcpp/expr.c
--- gcc-10.2.0/libcpp/expr.c	2020-07-23 00:35:18
+++ gcc-10.2.0-patched/libcpp/expr.c	2023-09-09 13:21:23
@@ -793,16 +793,16 @@
       if ((result & CPP_N_WIDTH) == CPP_N_LARGE
 	  && CPP_OPTION (pfile, cpp_warn_long_long))
         {
-          const char *message = CPP_OPTION (pfile, cplusplus) 
-				? N_("use of C++11 long long integer constant")
-		                : N_("use of C99 long long integer constant");
-
 	  if (CPP_OPTION (pfile, c99))
             cpp_warning_with_line (pfile, CPP_W_LONG_LONG, virtual_location,
-				   0, message);
+				   0, CPP_OPTION (pfile, cplusplus) 
+                   ? N_("use of C++11 long long integer constant")
+                   : N_("use of C99 long long integer constant"));
           else
             cpp_pedwarning_with_line (pfile, CPP_W_LONG_LONG,
-				      virtual_location, 0, message);
+				      virtual_location, 0, CPP_OPTION (pfile, cplusplus) 
+                      ? N_("use of C++11 long long integer constant")
+                      : N_("use of C99 long long integer constant"));
         }
 
       result |= CPP_N_INTEGER;
diff -ruN gcc-10.2.0/libcpp/macro.c gcc-10.2.0-patched/libcpp/macro.c
--- gcc-10.2.0/libcpp/macro.c	2020-07-23 00:35:18
+++ gcc-10.2.0-patched/libcpp/macro.c	2023-09-09 13:23:54
@@ -90,9 +90,6 @@
   node_type type;
 };
 
-static const char *vaopt_paste_error =
-  N_("'##' cannot appear at either end of __VA_OPT__");
-
 static void expand_arg (cpp_reader *, macro_arg *);
 
 /* A class for tracking __VA_OPT__ state while iterating over a
@@ -181,7 +178,7 @@
 	if (m_state == 2 && token->type == CPP_PASTE)
 	  {
 	    cpp_error_at (m_pfile, CPP_DL_ERROR, token->src_loc,
-			  vaopt_paste_error);
+              "'##' cannot appear at either end of __VA_OPT__");
 	    return ERROR;
 	  }
 	/* Advance states before further considering this token, in
@@ -210,7 +207,7 @@
 		if (was_paste)
 		  {
 		    cpp_error_at (m_pfile, CPP_DL_ERROR, token->src_loc,
-				  vaopt_paste_error);
+                  "'##' cannot appear at either end of __VA_OPT__");
 		    return ERROR;
 		  }
 
@@ -3361,8 +3358,6 @@
 create_iso_definition (cpp_reader *pfile)
 {
   bool following_paste_op = false;
-  const char *paste_op_error_msg =
-    N_("'##' cannot appear at either end of a macro expansion");
   unsigned int num_extra_tokens = 0;
   unsigned nparms = 0;
   cpp_hashnode **params = NULL;
@@ -3487,7 +3482,8 @@
 	     function-like macros, but not at the end.  */
 	  if (following_paste_op)
 	    {
-	      cpp_error (pfile, CPP_DL_ERROR, paste_op_error_msg);
+          cpp_error (pfile, CPP_DL_ERROR,
+            "'##' cannot appear at either end of a macro expansion");
 	      goto out;
 	    }
 	  if (!vaopt_tracker.completed ())
@@ -3502,7 +3498,8 @@
 	     function-like macros, but not at the beginning.  */
 	  if (macro->count == 1)
 	    {
-	      cpp_error (pfile, CPP_DL_ERROR, paste_op_error_msg);
+          cpp_error (pfile, CPP_DL_ERROR,
+            "'##' cannot appear at either end of a macro expansion");
 	      goto out;
 	    }
 
